src/main/java/com/google/api/generator/engine/ast/TypeNode.java

```diff
@@ -101,6 +101,14 @@ public enum TypeKind {
 
   public abstract boolean isArray();
 
+  public static TypeNode arrayOf(TypeNode type) {
```

Comment:
The name is a little confusing, sounds like it's creating a new array, I'm not good at naming as well, maybe something like `createArrayTypeOf()` or `ofArrayType()`?

Reply:
`createArrayTypeOf` seems more precise - I'll use that!